substrate/frame/asset-conversion/ops/src/mock.rs

```diff
@@ -0,0 +1,214 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Test environment for Asset Conversion Ops pallet.
+
+use super::*;
+use crate as pallet_asset_conversion_ops;
+use core::default::Default;
+use frame_support::{
+	construct_runtime, derive_impl,
+	instances::{Instance1, Instance2},
+	ord_parameter_types, parameter_types,
+	traits::{
+		tokens::{
+			fungible::{NativeFromLeft, NativeOrWithId, UnionOf},
+			imbalance::ResolveAssetTo,
+		},
+		AsEnsureOriginWithArg, ConstU128, ConstU32, ConstU64,
+	},
+	PalletId,
+};
+use frame_system::{EnsureSigned, EnsureSignedBy};
+use pallet_asset_conversion::{self, AccountIdConverter, AccountIdConverterNoSeed, Ascending};
+use sp_arithmetic::Permill;
+use sp_core::H256;
+use sp_runtime::{
+	traits::{AccountIdConversion, BlakeTwo256, IdentityLookup},
+	BuildStorage,
+};
+
+type Block = frame_system::mocking::MockBlock<Test>;
+
+construct_runtime!(
+  pub enum Test
+  {
+	System: frame_system,
+	Balances: pallet_balances,
+	Assets: pallet_assets::<Instance1>,
+	PoolAssets: pallet_assets::<Instance2>,
+	AssetConversion: pallet_asset_conversion,
+	AssetConversionOps: pallet_asset_conversion_ops,
+  }
+);
+
+#[derive_impl(frame_system::config_preludes::TestDefaultConfig as frame_system::DefaultConfig)]
```

Comment:
You can remove many types given that you're deriving the defaults. Take a look at: https://github.com/paritytech/polkadot-sdk/blob/4b5c3fd0cbb47f3484671ffce284b7586311126a/substrate/frame/balances/src/tests/mod.rs#L93

Reply:
updated